zap_otlp_sync/zap_otlp_sync.go

```diff
@@ -0,0 +1,184 @@
+package zap_otlp_sync
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	collpb "go.opentelemetry.io/proto/otlp/collector/logs/v1"
+	cv1 "go.opentelemetry.io/proto/otlp/common/v1"
+	lpb "go.opentelemetry.io/proto/otlp/logs/v1"
+	rv1 "go.opentelemetry.io/proto/otlp/resource/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+type OtelSyncer struct {
+	ctx       context.Context
+	close     context.CancelFunc
+	res       *rv1.Resource
+	resSchema string
+	scope     *cv1.InstrumentationScope
+	values    [][]byte
+	queue     chan []byte
+
+	client        collpb.LogsServiceClient
+	batchSize     int
+	sendBatch     chan bool
+	closeExporter chan bool
+	valueMutex    sync.Mutex
+}
+
+type Options struct {
+	BatchSize          int
+	BatchIntervalInSec int
+	ResourceSchema     string
+	Scope              *instrumentation.Scope
+	Resource           *resource.Resource
+}
+
+func NewOtlpSyncer(conn *grpc.ClientConn, options Options) *OtelSyncer {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if options.BatchSize == 0 {
+		options.BatchSize = 100
+	}
+
+	if options.BatchIntervalInSec == 0 {
+		options.BatchIntervalInSec = 5
+	}
+
+	var rattrs *rv1.Resource
+	if options.Resource != nil {
+		iter := options.Resource.Iter()
+		attrs := make([]*cv1.KeyValue, 0, iter.Len())
+		for iter.Next() {
+			attr := iter.Attribute()
+			attrs = append(attrs, &cv1.KeyValue{
+				Key:   string(attr.Key),
+				Value: &cv1.AnyValue{Value: &cv1.AnyValue_StringValue{StringValue: attr.Value.Emit()}},
+			})
+		}
+		rattrs = &rv1.Resource{Attributes: attrs}
+	}
+
+	var instrumentationScope *cv1.InstrumentationScope
+	if options.Scope != nil {
+		instrumentationScope = &cv1.InstrumentationScope{
+			Name:    options.Scope.Name,
+			Version: options.Scope.Version,
+		}
+	}
```

Comment:
This should be removed in favour of using logger name for scope

Reply:
Have added this changes, to use a map for instrumentation scope using logger name